Add Autocomplete api

kennyDang opened the pull request:
Removed app id and secret

kennyDang committed:
Added autocomplete feature to Yelp’s API

kennyDang committed:
Added unit testing for autocomplete

kennyDang committed:
Removed client secret and app id

kennyDang committed:
Accidentally included app id and secret

mittonk commented:
@d9chen , @SSheldon --- thoughts?

d9chen commented on Classes/Client/YLPClient+Autocomplete.m:
```diff
@@ -0,0 +1,51 @@
+@implementation YLPClient (Autocomplete)
+
+- (NSURLRequest *)autoCompleteRequestWithParams:(NSDictionary *)params {
+    NSString *phoneSearchPath = @"/v3/autocomplete";
```
phoneSearchPath -> autoCompletePath

d9chen commented on Classes/Client/YLPClient+Autocomplete.h:
```diff
@@ -0,0 +1,28 @@
+@interface YLPClient (Autocomplete)
+
+- (void)fetchAutocompleteSuggestionsWithTerm:(NSString *)term
+                                  coordinate:(YLPCoordinate *)coordinate
```
Seems like this is also optional. Could also add nullable for this param.

d9chen commented on Classes/Response/YLPAutocomplete.m:
```diff
@@ -0,0 +1,54 @@
+}
+
++ (NSArray *)termsFromJSONArray:(NSArray *)termsJSON {
+    NSMutableArray<NSString *> *mutableTermsJSON = [[NSMutableArray alloc] init];
```
This should probably be an array of dictionaries to accurately represent the return type of terms from the API.

d9chen commented on YelpAPITests/Client/YLPAutocompleteClientTestCase.m:
```diff
@@ -0,0 +1,183 @@
+    NSDictionary *expectedSecondCategory = expectedResponse[@"categories"][1];
+    NSDictionary *expectedThirdCategory = expectedResponse[@"categories"][2];
+    
+    double latitude = 37.7613570;
```
Since this isn't a required parameter for a category autocomplete search can we also add a test case for when latitude/longitude aren't provided. (once the code supports it)

kennyDang replied:
Thank you for reviewing my pull request! I will make the correct changes and send an updated pull request once I get home :)

d9chen reviewed (commented):
Took a first pass. Thanks for adding this in @kennyDang !

kennyDang committed:
Renamed phoneSearchPath to autoCompletePath

kennyDang committed:
Made corrections

- Changed phoneSearchPath to autoCompletePath
- Changed coordinate parameter to an optional
- Changed mutableTermsJSON into an array of dictionary
- Added a new test for autocomplete search when no coordinate exists

kennyDang commented:
@d9chen @SSheldon I just finished making the corrections (hopefully)

kennyDang commented:
@d9chen @SSheldon Has there been any updates for me?

kennyDang commented:
@SSheldon @d9chen Hey guys, I know y'all are busy but I'm just checking in. Has there been any progress? Thanks!

d9chen reviewed (approved):
Sorry, about the long turn around time @kennyDang, but this change looks good to me!

dan98765 commented:
Thanks for contributing, @kennyDang

kennyDang commented:
@dan98765 Thank you for merging in the changes. I forgot about this haha.
